airbyte-ci/connectors/pipelines/pipelines/actions/environments.py

```diff
@@ -302,29 +331,25 @@ async def with_installed_python_package(
     Returns:
         Container: A python environment container with the python package installed.
     """
-    install_requirements_cmd = ["python", "-m", "pip", "install", "-r", "requirements.txt"]
-    install_connector_package_cmd = ["python", "-m", "pip", "install", "."]
-
     container = with_python_package(context, python_environment, package_source_code_path, exclude=exclude)
 
     local_dependencies = await find_local_python_dependencies(context, package_source_code_path)
 
     for dependency_directory in local_dependencies:
         container = container.with_mounted_directory("/" + dependency_directory, context.get_repo_dir(dependency_directory))
 
-    has_setup_py, has_requirements_txt = await check_path_in_workdir(container, "setup.py"), await check_path_in_workdir(
-        container, "requirements.txt"
-    )
+    has_setup_py = await check_path_in_workdir(container, "setup.py")
+    has_requirements_txt = await check_path_in_workdir(container, "requirements.txt")
+    has_pyproject_toml = await check_path_in_workdir(container, "pyproject.toml")
+
+    if has_pyproject_toml:
+        container = _install_python_dependencies_from_poetry(container)
 
     if has_setup_py:
-        container = container.with_exec(install_connector_package_cmd)
-    if has_requirements_txt:
-        container = container.with_exec(install_requirements_cmd)
+        container = _install_python_dependencies_from_setup_py(container, additional_dependency_groups)
 
-    if additional_dependency_groups:
-        container = container.with_exec(
-            install_connector_package_cmd[:-1] + [install_connector_package_cmd[-1] + f"[{','.join(additional_dependency_groups)}]"]
-        )
+    if has_requirements_txt:
+        container = _install_python_dependencies_from_requirements_txt(container)
```

Comment:
@bnchrch would you mind adding a unit test for this?

Reply:
@alafanechere My change will add!